x-pack/plugins/cases/public/components/actions/assignees/edit_assignees_flyout.test.tsx

```diff
@@ -0,0 +1,93 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import React from 'react';
+import userEvent from '@testing-library/user-event';
+import type { AppMockRenderer } from '../../../common/mock';
+import { createAppMockRenderer } from '../../../common/mock';
+import { basicCase } from '../../../containers/mock';
+import { waitForComponentToUpdate } from '../../../common/test_utils';
+import { EditAssigneesFlyout } from './edit_assignees_flyout';
+import { waitFor } from '@testing-library/react';
+
+jest.mock('../../../containers/user_profiles/api');
+
+describe('EditAssigneesFlyout', () => {
+  let appMock: AppMockRenderer;
+
+  /**
+   * Case one has the following assignees: coke, pepsi, one
+   * Case two has the following assignees: one, three
+   * All available assignees are: one, two, three, coke, pepsi
+   */
+  const props = {
+    selectedCases: [basicCase],
+    onClose: jest.fn(),
+    onSaveAssignees: jest.fn(),
+  };
+
+  beforeEach(() => {
+    appMock = createAppMockRenderer();
+    jest.clearAllMocks();
+  });
+
+  it('renders correctly', async () => {
+    const result = appMock.render(<EditAssigneesFlyout {...props} />);
+
+    expect(result.getByTestId('cases-edit-assignees-flyout')).toBeInTheDocument();
+    expect(result.getByTestId('cases-edit-assignees-flyout-title')).toBeInTheDocument();
+    expect(result.getByTestId('cases-edit-assignees-flyout-cancel')).toBeInTheDocument();
+    expect(result.getByTestId('cases-edit-assignees-flyout-submit')).toBeInTheDocument();
+
+    await waitForComponentToUpdate();
+  });
+
+  it('calls onClose when pressing the cancel button', async () => {
+    const result = appMock.render(<EditAssigneesFlyout {...props} />);
+
+    userEvent.click(result.getByTestId('cases-edit-assignees-flyout-cancel'));
+    expect(props.onClose).toHaveBeenCalled();
+
+    await waitForComponentToUpdate();
+  });
+
+  it('calls onSaveTags when pressing the save selection button', async () => {
+    const result = appMock.render(<EditAssigneesFlyout {...props} />);
+
+    await waitForComponentToUpdate();
+
+    await waitFor(() => {
+      expect(result.getByText('Damaged Raccoon')).toBeInTheDocument();
+    });
+
+    userEvent.click(result.getByText('Damaged Raccoon'));
+    userEvent.click(result.getByTestId('cases-edit-assignees-flyout-submit'));
+
+    expect(props.onSaveAssignees).toHaveBeenCalledWith({
+      selectedItems: [],
+      unSelectedItems: ['u_J41Oh6L9ki-Vo2tOogS8WRTENzhHurGtRc87NgEAlkc_0'],
+    });
+  });
+
+  it('shows the case title when selecting one case', async () => {
+    const result = appMock.render(<EditAssigneesFlyout {...props} />);
+
+    expect(result.getByText(basicCase.title)).toBeInTheDocument();
```

Comment:
nit: The recommended way by the testing library is actually to use `getByRole` for this kind of thing.

```suggestion
    expect(result.getByRole('heading', {name:basicCase.title})).toBeInTheDocument();
```

One of the things I found out during the ON Week when trying to get rid of our warnings. 😄

Reply:
Thanks! Curious why this eliminates the warnings.

Reply:
Ah no no. I meant I read about it when doing my warnings ON week project, it doesn't actually eliminate any.

Among other things, it provides a better output when a test fails if things don't render correctly.

Reply:
Oh, nice! Thanks!

Reply:
I tested and it is not working. The problem is that the title is not a heading but a text in a paragraph. Do you mind if I leave it as it is? I did not find a better query that  `getByText`.

Reply:
yup you can leave it 👍